### PR TITLE
Add core DevAssist class and shared types

### PR DESCRIPTION
**Pull request by hdprajwal**
Introduces API integration, mainly focusing on adding support for multiple AI providers and ensuring smooth interoperability between different message formats. The most important changes include updates to the `package.json` to add new dependencies, the creation of handlers for new AI providers, and the addition of transformation functions to convert messages between different formats.

### Dependency Updates:
* Added dependencies for `@anthropic-ai/sdk`, `@google/generative-ai`, and `openai` to support new AI providers.

### API Handler Additions:
* Introduced a factory function `createApiHandler` to instantiate handlers for different AI providers based on configuration.
* Implemented `AnthropicHandler` to handle API interactions with Anthropic AI.
* Implemented `GeminiHandler` to handle API interactions with Google Gemini.
* Implemented `OllamaHandler` to handle API interactions with Ollama AI.
* Implemented `OpenAiHandler` to handle API interactions with OpenAI.

### Message Transformation Functions:
* Added functions to convert messages and content between Anthropic and Google Gemini formats.
* Added functions to convert messages and content between Anthropic and OpenAI formats.

### CODE REVIEW

File: src/shared/api.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+export type ApiProvider = "ollama" | "openai" | "gemini" | "anthropic";
```
Comment (adithya1012) [inline src/shared/api.ts:1]: These are LLMs we are supporting and in this Ollama and Gemini are freely available. So do we need to include all other LLMs also ?

**Comment by adithya1012** [comment]
We should have a knowledge sharing session on this API integration with different LLMs.

**Comment by hdprajwal** [comment]
Sure, Lets connect tomorrow over Teams to go over the API integrations.

**Review by adithya1012 — APPROVED** [review]

File: src/shared/api.ts (hunk 1 above)
Comment (hdprajwal) [reply src/shared/api.ts:1]: When I try to run Ollama on my local machine its relatively slow and gemini has a request limit. So I have added two other LLM ie. OpenAI and Anthropic for us to test out the performance benifits. We can utilize Ollama or Gemini once we are done developing the extension.

**Review by jahnavib26 — APPROVED** [review]

**Review by nidhitalastha — APPROVED** [review]
This looks good to me, I have added the basic Webview code to this PR. 
Will create another PR for my part of the integration code.

**Review by Sinduja98 — APPROVED** [review]
It looks good. It's running fine on my local.

**Comment by jahnavib26** [comment]
Works fine on my local.

**Pull request merged**